model/flow/executionResult.go

```diff
@@ -23,3 +22,13 @@ func (er ExecutionResult) ID() Identifier {
 func (er ExecutionResult) Checksum() Identifier {
 	return MakeID(er)
 }
+
+// FinalStateCommit gets the final state of the result
+// if the number of chunks are 0 returns empty statecommit
+func (er ExecutionResult) FinalStateCommit() StateCommitment {
+	if er.Chunks.Len() == 0 {
+		return StateCommitment{}
```

Comment:
When chunks is empty, the final state commit would be the previous result's final statecommit, so it's not actuate to return an empty statecommit here. 

However, an execution result doesn't have a directly link to the previous result's statecommit. Therefore, I think we might want to change the `FinalStateCommit` to return `(StateCommitment, bool)`. And when there is no chunk, we return `StateCommitment{}, false`. So that the business logic that calls the `FinalStateCommmit` will decide what to do when the result has no chunk.

For instance, if the business logic considers a certain result should have at least one chunk, then the returned bool value from the `FinalStateCommit` method could be used for sanity check.

Reply:
I agree that state commitment cannot be empty and should reference previous one.
However, `ExecutionResult` has a field `PreviousResultID Identifier // commit of the previous ER` which creates a link to previous `ER` and therefore previous `FinalStateCommit`.